tests/test_update_query.py

```diff
@@ -117,6 +117,45 @@ def test_with_query_list_int():
     assert str(url.with_query([("a", 1)])) == "http://example.com/?a=1"
 
 
+def test_with_query_sequence():
+    url = URL("http://example.com")
+    assert url.with_query({"a": [1, 2]}) == URL("http://example.com/?a=1&a=2")
+
+
+def test_with_query_sequence_tuple():
+    url = URL("http://example.com")
+    assert url.with_query({"a": (1, 2)}) == URL("http://example.com/?a=1&a=2")
+
+
+def test_with_query_sequence_mixed_types():
+    url = URL("http://example.com")
+    assert url.with_query({"a": ["1", 2]}) == URL("http://example.com/?a=1&a=2")
+
+
+def test_with_query_sequence_single_then_list():
+    url = URL("http://example.com")
+    expected = URL("http://example.com/?a=1&b=2&b=3")
+    assert url.with_query({"a": 1, "b": [2, 3]}) == expected
+
+
+def test_with_query_sequence_list_then_single():
+    url = URL("http://example.com")
+    expected = URL("http://example.com/?a=1&a=2&b=3")
+    assert url.with_query({"a": [1, 2], "b": 3}) == expected
```

Comment:
these tests are essentially one sigle test and so it should use `@pytest.mark.parametrize`. You can use `pytest.param(id=)` if you want to improve the test report.

Reply:
I'm not that familiar with `pytest`, and none of the previous test used that (although it seems they could), so that's why I didn't. But I will make the changes.